include/mesh.tcc

```diff
@@ -501,7 +502,8 @@ bool mpm::Mesh<Tdim>::generate_material_points(unsigned nquadratures,
           status = this->add_particle(particle, checks);
           if (status) {
             map_particles_[pid]->assign_cell(*citr);
-            map_particles_[pid]->assign_material(material);
+            for (unsigned phase = 0; phase < material.size(); phase++)
+              map_particles_[pid]->assign_material(material[phase], phase);
```

Comment:
how can we check if the particle can take more than 1 material?

Reply:
That's a good point, will that be better to assign an `unsigned` in particle which denotes the `phase_size` it contains? I will then use `map_particles_[pid]->phase_size();` instead of `material.size()`.

Reply:
Could we just store an internal variable, and throw if someone tries to add excess particle?

Reply:
I am not sure what do you mean by an index-based approach of adding material?

Reply:
So should I store an internal variable to denote the `phase_size`? There, I can assert if the `phase` passed in `assign_material` is larger or equal to the `phase_size`.

Reply:
In the constructor can we do:
```
// for 2 phase
materials_.resize(2);
```
and then later in the assign_material we just do
```
materials_.at(phase) = material;
```
this way if we access an invalid phase, c++ `.at()` will throw.

Reply:
Instead of putting 2 or 1 in resize, I think it will be better to call it from a function `phase_size()` or a member variable `phase_size_`, there we don't have to write the same line of code again.

Reply:
That would require an extra variable and a function. While setting materials size in constructor does the same thing.

Reply:
Okay, as for now, I think your idea works better. I was initially thinking to reuse `phase_size()` for two-phase two-points coupled but if I need it, I will modify it later.

Reply:
@kks32 I decided to write a protected function to resize. This can be used for the derived particle classes too and it is less messy.